ntbs-service/DataMigration/MigrationRepository.cs

```diff
@@ -256,6 +273,16 @@ public async Task<IEnumerable<MigrationDbMBovisMilkConsumption>> GetMigrationMBo
             }
         }
 
+        public async Task<IEnumerable<MigrationLegacyUser>> GetLegacyUserByUsername(string username)
```

Comment:
There's a mismatch between the method name and the return type here - do we want to call `Single` or `SingleOrDefault` here?